File: scripts/create_github_issue.js
const { Octokit } = require("@octokit/rest");
require('dotenv').config();

const octokit = new Octokit({
  auth: process.env.GITHUB_TOKEN,
});

async function createIssue(owner, repo, title, body, labels = [], assignees = []) {
  try {
    const response = await octokit.issues.create({
      owner,
      repo,
      title,
      body,
      labels,
      assignees,
    });
    console.log(`Issue created: ${response.data.html_url}`);
  } catch (error) {
    console.error(`Error creating issue: ${error.message}`);
  }
}

function main() {
  const args = process.argv.slice(2);
  if (args.length < 4) {
    console.error("Usage: node create_github_issue.js <owner> <repo> <title> <body> [labels] [assignees]");
    process.exit(1);
  }

  const [owner, repo, title, body, labels, assignees] = args;

  createIssue(
    owner,
    repo,
    title,
    body,
    labels ? labels.split(',') : [],
    assignees ? assignees.split(',') : []
  );
}

main();

// Example usage:
// node create_github_issue.js BigOleHealz ContractAnalyzer "Test Issue" "This is a test issue created programmatically." "bug,help wanted" "username1,username2"

// Note:
// - Ensure you have a .env file with GITHUB_TOKEN set to your personal access token.
// - Labels and assignees are optional and should be comma-separated if provided.
// - This script requires the @octokit/rest package. Install it using npm or yarn.
// - Run `npm install dotenv @octokit/rest` to install the necessary packages.

File: scripts/create_github_issue.py
import requests
import os

# Function to create a GitHub issue
def create_github_issue(repo, title, body=None, labels=None, assignees=None):
    url = f"https://api.github.com/repos/{repo}/issues"
    headers = {
        "Authorization": f"token {os.getenv('GITHUB_TOKEN')}",
        "Accept": "application/vnd.github.v3+json"
    }
    issue = {
        "title": title,
        "body": body,
        "labels": labels or [],
        "assignees": assignees or []
    }
    response = requests.post(url, json=issue, headers=headers)
    if response.status_code == 201:
        print(f"Successfully created issue {title}")
    else:
        print(f"Failed to create issue {title}: {response.content}")

# Example usage
if __name__ == "__main__":
    repo_name = "owner/repo"  # Replace with your repository
    issue_title = "Test Issue"
    issue_body = "This is a test issue created by the script."
    issue_labels = ["bug"]
    issue_assignees = ["username"]
    create_github_issue(repo_name, issue_title, issue_body, issue_labels, issue_assignees)
